cube/cube.cpp

```diff
@@ -511,76 +512,17 @@ static void registry_handle_global_remove(void *data, wl_registry *registry, uin
 static const wl_registry_listener registry_listener = {registry_handle_global, registry_handle_global_remove};
 #endif
 
-Demo::Demo()
-    :
-#if defined(VK_USE_PLATFORM_WIN32_KHR)
-      connection{nullptr},
-      window{nullptr},
-      minsize(POINT{0, 0}),  // Use explicit construction to avoid MSVC error C2797.
-#endif
+Demo::Demo() {
+    name = APP_SHORT_NAME;
 
-#if defined(VK_USE_PLATFORM_XLIB_KHR)
-      xlib_window{0},
-      xlib_wm_delete_window{0},
-      display{nullptr},
-#elif defined(VK_USE_PLATFORM_XCB_KHR)
-      xcb_window{0},
-      screen{nullptr},
-      connection{nullptr},
-#elif defined(VK_USE_PLATFORM_WAYLAND_KHR)
-      display{nullptr},
-      registry{nullptr},
-      compositor{nullptr},
-      window{nullptr},
-      wm_base{nullptr},
-      xdg_decoration_mgr{nullptr},
-      toplevel_decoration{nullptr},
-      window_surface{nullptr},
-      xdg_surface_has_been_configured{false},
-      window_toplevel{nullptr},
-      seat{nullptr},
-      pointer{nullptr},
-      keyboard{nullptr},
-#elif defined(VK_USE_PLATFORM_DIRECTFB_EXT)
-      dfb{nullptr},
-      window{nullptr},
-      event_buffer{nullptr},
-#endif
-      prepared{false},
-      use_staging_buffer{false},
-      use_xlib{false},
-      graphics_queue_family_index{0},
-      present_queue_family_index{0},
-      enabled_extension_count{0},
-      enabled_layer_count{0},
-      width{0},
-      height{0},
-      swapchainImageCount{0},
-      presentMode{vk::PresentModeKHR::eFifo},
-      frame_index{0},
-      spin_angle{0.0f},
-      spin_increment{0.0f},
-      pause{false},
-      quit{false},
-      curFrame{0},
-      frameCount{0},
-      validate{false},
-      use_break{false},
-      suppress_popups{false},
-      force_errors{false},
-      current_buffer{0},
-      queue_family_count{0} {
-#if defined(VK_USE_PLATFORM_WIN32_KHR)
-    memset(name, '\0', APP_NAME_STR_LEN);
-#endif
     memset(projection_matrix, 0, sizeof(projection_matrix));
     memset(view_matrix, 0, sizeof(view_matrix));
     memset(model_matrix, 0, sizeof(model_matrix));
 }
 
-void Demo::build_image_ownership_cmd(uint32_t const &i) {
-    auto const cmd_buf_info = vk::CommandBufferBeginInfo().setFlags(vk::CommandBufferUsageFlagBits::eSimultaneousUse);
-    auto result = swapchain_image_resources[i].graphics_to_present_cmd.begin(&cmd_buf_info);
+void Demo::build_image_ownership_cmd(SwapchainImageResources const &swapchain_image_resource) {
+    auto result = swapchain_image_resource.graphics_to_present_cmd.begin(
```

Comment:
These two commands switched order.  It's not clear why.

Reply:
They aren't switched order. I am instead creating the createInfo struct inside the function call instead of outside, because I can. (since it takes it by reference not pointer).